src/core_plugins/kbn_vislib_vis_types/public/controls/tooltip_options.js

```diff
@@ -0,0 +1,221 @@
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import React from 'react';
+import Select from 'react-select';
+
+const visTypeExcludeList = [
+  'input_control_vis',
+  'tagcloud',
+  'tile_map',
+  'timelion',
+  'region_map',
+  'metrics' // TSVB
+];
+
+export function TooltipOptions(props) {
+
+  const findVisualizations = async (input) => {
+    const findOptions = {
+      type: ['visualization'],
+      fields: ['title', 'visState'],
+      search: `${input}*`,
+      search_fields: ['title^3', 'description'],
+      page: 1,
+      perPage: 100
+    };
+
+    let options = [];
+    let fetchNextPage = false;
+
+    do {
+      const resp = await props.savedObjectsClient.find(findOptions);
+      const optionsFromPage = resp.savedObjects
+      .filter((savedObject) => {
```

Comment:
I would also filter out the visualization itself. 

This only occurs if the visualization is already saved, but it shows up in the list. It doesn't really add anything.

e.g.:
![image](https://user-images.githubusercontent.com/1833023/31002699-2412847a-a4ba-11e7-9cb0-ea2e74f22037.png)